mm2src/coins/z_coin/storage/walletdb/wasm/storage.rs

```diff
@@ -1083,7 +1083,7 @@ impl WalletRead for WalletIndexedDb {
             let matching_tx = maybe_txs.iter().find(|(id_tx, _tx)| id_tx.to_bigint() == note.spent);
 
             if let Some((_, tx)) = matching_tx {
-                if tx.block.is_none() {
+                if tx.block.is_some() {
```

Comment:
This changes the behaviour significantly, do we have a test coverage for this?

Reply:
yep, there's a test here, https://github.com/KomodoPlatform/komodo-defi-framework/blob/07bd8f00900bed92cbdf156ea87bf875a5f52075/mm2src/coins/z_coin/storage/walletdb/wasm/mod.rs#L1076 .. only tx that's been mined or included in a block can have a nullifier.